src/Source/DatabaseSource.php

```diff
@@ -51,7 +52,14 @@ public function getMappedData(): array
         $db = Database::getInstance($sourceModel->row());
         $where = $sourceModel->dbSourceTableWhere ?: '1=1';
 
-        $records = $db->prepare("SELECT * FROM $sourceModel->dbSourceTable WHERE $where")->execute();
+        try {
+            $records = $db->prepare("SELECT * FROM $sourceModel->dbSourceTable WHERE $where")->execute();
+        } catch (\Exception $e) {
+            // catch and throw new exception to specify error message
+            Controller::loadLanguageFile('default');
+
+            throw new \Exception(sprintf($GLOBALS['TL_LANG']['MSC']['entityImport']['dbConnectionError'], $e->getMessage()));
+        }
```

Comment:
`$db->getInstance()` also needs to by in try, doesn't it? An error might be thrown on connection already.

Reply:
From backend config it would be prevented since the relevant  fields are mandatory. But there certainly could be a misconfiguration if eg it is edited directly through the database. I added a try-catch block